source/common/modules/markdown-editor/commands/autocorrect.ts

```diff
@@ -35,12 +35,20 @@ const startChars = ' ([{-–—\n\r\t\v\f'
  */
 function posInProtectedNode (state: EditorState, pos: number): boolean {
   const node = syntaxTree(state).resolve(pos, 0)
+
+  // Having the cursor at the end of the HorizontalRule '---' is not considered inside the HorizontalRule node
+  // So we check the previous position to see if it is in a HorizontalRule node
+  const checkYAMLFrontmatterNode = syntaxTree(state).resolve(pos - 1, 0)
+  const checkYAMLFrontmatter = [ 'YAMLFrontmatterStart', 'YAMLFrontmatterEnd', 'HorizontalRule' ].includes(checkYAMLFrontmatterNode.type.name)
+  if (checkYAMLFrontmatter) {
+    return checkYAMLFrontmatter
```

Comment:
If my above comment works (see the same function call on line 37, I think it should work if you replace the second parameter with `-1`, but again, please check the docs first), then you can remove this entire check and add those nodes below to the array of protected nodes, cleaning this up a little.